lib/auth/server_info.go

```diff
@@ -88,6 +88,12 @@ func getServerInfoNames(node types.Server) []string {
 
 func (a *Server) setLabelsOnNodes(ctx context.Context, nodes []types.Server) (failedUpdates int, err error) {
 	for _, node := range nodes {
+		// EICE Node labels can't be updated using the Inventory Control Stream because there's no reverse tunnel.
+		// Labels are updated by the DiscoveryService during 'Server.handleEC2Instances'.
+		if node.GetSubKind() == types.SubKindOpenSSHEICENode {
+			continue
+		}
```

Comment:
What about OpenSSH nodes? They also don't have reverse tunnel but they aren't being skipped here. Just trying to understand why this difference between "subkind: openssh" and "subkind: eic" nodes.

Reply:
I don't think it works for OpenSSH nodes, and it probably gets reported as
```
unable to update labels on %v node(s) due to missing control stream
```

I'll do some tests and get back to you

Reply:
I don't think we support ServerInfo for Agentless nodes.
There's at least two issues.
One of them was reported here: https://github.com/gravitational/teleport/issues/38035

The other one is as follows:
When running the agentless script, labels are computed based on the instance's metadata service endpoint.
We then run `teleport join openssh` and the node is added to teleport with the following content:
```yaml
kind: node
metadata:
  labels:
    teleport.dev/account-id: "<account-id>"
    teleport.dev/aws-region: eu-west-2
    teleport.dev/instance-id: <i-nstance_id>
  name: <uuid>
spec:
  addr: <addr>:22
  hostname: ip-.eu-west-2.compute.internal
  version: ""
sub_kind: openssh
version: v2
```

On the DiscoveryService, a ServerInfo is created as well with the following:
```yaml
kind: server_info
metadata:
  name: aws-<account_id>-<i-nstance_id>
spec:
  new_labels:
    aws/AnotherTag: AnotherTagValue
    aws/Discover: YesPlease
    aws/Name: MarcoAgentlessBox
sub_kind: cloud_info
version: v1
```

However, when calculating the corresponding ServerInfo for the node we are trying to compare against two different values
https://github.com/gravitational/teleport/blob/e942e5fb743731055dceefe76b0a2f98ff73b3c1/lib/auth/server_info.go#L79-L87
We lookup the ServerInfo using the Node's name because the Node doesn't have AWS CloudMetadata, as we can see above.
It doesn't exist because the corresponding ServerInfo's name is composed using the AWS Account ID and the EC2 InstanceID.

However, even if it existed, we would not be able to update their labels because it requires a reverse tunnel connection, which is not even possible with OpenSSH agentless nodes.

---
Getting back to your question.
I think we should skip both subkinds here: `openssh` and `openssh-eice`.

cc @lxea

Reply:
We had the very same discussion a week ago: https://gravitational.slack.com/archives/C0DF0TPMY/p1707155794572529?thread_ts=1707151022.774659&cid=C0DF0TPMY

Server info is here to allow reconfiguring labels without restarting the agent. This is unnecessary on any agentless node and gets ion the way of IAC/any kind of automation. Server info should only apply if the subkind is `teleport`